### PR TITLE
Polishing backend module

### PR DESCRIPTION
**Pull request by rreiner-hqs**

### CODE REVIEW

File: roqoqo-quest/src/backend.rs [hunk 1]
```diff
@@ -89,11 +98,13 @@ impl Backend {
...
     /// Sets the number of repetitions used for stochastic circuit simulations
     ///
-    /// The number of repetitions of the actual simulation is set to one by default.
-    /// The repetitions are not to be confused with the number of simulated measurements per simulation run
-    /// set with PragmaRepeatedMeasurement or PragmaSetNumberMeasurements.
-    /// Should only be different from one if a stochastic unravelling of a noisy simulation is used
-    /// with PragmaRandomNoise or PragmaOverrotation
+    /// The number of repetitions of the actual simulation is set to one by default. The repetitions
+    /// are not to be confused with the number of simulated measurements per simulation run, which
+    /// is set with PragmaRepeatedMeasurement or PragmaSetNumberMeasurements. Should only be
```
Comment (dgupta-hqs) [inline roqoqo-quest/src/backend.rs:103]:
```suggestion
    /// is set with PragmaRepeatedMeasurement or PragmaSetNumberMeasurements. It should only be
```

File: roqoqo-quest/src/backend.rs [hunk 2]
```diff
@@ -333,22 +311,29 @@ impl Backend {
...
                 }
                 Operation::PragmaRepeatedMeasurement(o) => {
+                    // If we already have a repeated measurement
                     if let Some(nm) = number_measurements {
                         // Construct involved qubits
                         let involved_qubits: Vec<usize> = match register_lengths.get(o.readout()) {
-                            Some(pragma_nm) => {
-                                let n = *pragma_nm;
+                            // TODO what is this way of determining the involved qubits for a PRM?
+                            // The involved_qubits method for PRM returns All.
```
Comment (dgupta-hqs) [inline roqoqo-quest/src/backend.rs:319]: I believe the way to do that is to find the max qubit by iterating through the entire circuit.

File: roqoqo-quest/src/backend_refactor.rs [hunk 3]
```diff
@@ -0,0 +1,781 @@
...
+    /// `Option<Vec<u64>>` - The current random seed
+    pub fn get_random_seed(&self) -> Option<Vec<u64>> {
+        self.random_seed.clone()
+    }
+
+    /// Sets the number of repetitions used for stochastic circuit simulations
+    ///
+    /// The number of repetitions of the actual simulation is set to one by default. The repetitions
+    /// are not to be confused with the number of simulated measurements per simulation run, which
+    /// is set with PragmaRepeatedMeasurement or PragmaSetNumberMeasurements. Should only be
```
Comment (dgupta-hqs) [inline roqoqo-quest/src/backend_refactor.rs:103]:
```suggestion
    /// is set with PragmaRepeatedMeasurement or PragmaSetNumberMeasurements. It should only be
```

File: roqoqo-quest/src/interface/pragma_operations.rs [hunk 4]
```diff
@@ -163,34 +163,21 @@ pub fn execute_pragma_set_state_vector(
     let statevec = operation.statevector();
     let num_amps: i64 = statevec.len() as i64;
     if num_amps != 2_i64.pow(qureg.number_qubits()) {
-        return Err(RoqoqoBackendError::GenericError{
+        return Err(RoqoqoBackendError::GenericError {
             msg: format!(
-                "Can not set statevector: number of qubits of statevector ({}) differs from number of qubits in qubit register ({}).",
+                "Can not set statevector: number of qubits of statevector ({}) differs from number \
```
Comment (dgupta-hqs) [inline roqoqo-quest/src/interface/pragma_operations.rs:168]:
```suggestion
                "Cannot set statevector: number of qubits of statevector ({}) differs from number \
```

**Review by dgupta-hqs — APPROVED** [review]
Left a couple of minor suggestions.

File: roqoqo-quest/src/backend.rs (hunk 2 above)
Comment (rreiner-hqs) [reply roqoqo-quest/src/backend.rs:319]: I agree, in fact there is already a suggestion for this functionality in qoqo. In this inline comment I meant something else, namely why do we define the involved qubits in a PragmaRepeatedMeasurement by just looking at the length of the output register. Looking at it now it is clearer though, I'll remove the inline comment.

Comment (rreiner-hqs) [inline roqoqo-quest/src/backend.rs:319]:
```suggestion
```

**Pull request merged**